src/commands/lightning/preview/app.ts

```diff
@@ -116,22 +192,116 @@ export default class LightningPreviewApp extends SfCommand<void> {
     //
     // Here we've chosen the second approach
     const idx = this.argv.findIndex((item) => item.toLowerCase() === '-o' || item.toLowerCase() === '--target-org');
+    let targetOrg: string | undefined;
     if (idx >= 0 && idx < this.argv.length - 1) {
-      orgOpenCommandArgs.push('--target-org', this.argv[idx + 1]);
+      targetOrg = this.argv[idx + 1];
     }
 
-    await this.config.runCommand('org:open', orgOpenCommandArgs);
+    const launchArguments = PreviewUtils.generateDesktopPreviewLaunchArguments(ldpServerUrl, appId, targetOrg);
+
+    await this.config.runCommand('org:open', launchArguments);
   }
 
-  // eslint-disable-next-line class-methods-use-this, @typescript-eslint/no-unused-vars
-  private async iosPreview(logger: Logger, flags: LightningPreviewAppFlags): Promise<void> {
-    logger.info('Preview on iOS Not Implemented Yet');
-    return Promise.reject(new Error('Preview on iOS Not Implemented Yet'));
+  private async mobilePreview(
+    platform: Platform.ios | Platform.android,
+    ldpServerUrl: string,
+    appName?: string,
+    appId?: string,
+    deviceId?: string,
+    logger?: Logger
+  ): Promise<void> {
+    try {
+      // 1. Verify that user environment is set up for mobile (i.e. has right tooling)
+      await this.verifyMobileRequirements(platform, logger);
+
+      // 2. Fetch the target device
+      const device = await PreviewUtils.getMobileDevice(platform, deviceId, logger);
+      if (!device) {
+        throw new Error(messages.getMessage('error.device.notfound', [deviceId ?? '']));
+      }
+
+      // 3. Boot the device if not already booted
+      this.spinner.start(messages.getMessage('spinner.device.boot', [device.toString()]));
+      const resolvedDeviceId = platform === Platform.ios ? (device as IOSSimulatorDevice).udid : device.name;
+      const emulatorPort = await PreviewUtils.bootMobileDevice(platform, resolvedDeviceId, logger);
+      this.spinner.stop();
+
+      // 4. Generate self-signed certificate and wait for user to install it
+      // TODO: update the save location to be the same as server config file path
```

Comment:
That actually poses a good question. Given that the user needs to interact with the cert (i.e. for manual installation), it rules out putting it in a temp directory. I guess in the absence of the user being able to specify, I'd be inclined (in the final form) to simply save it in the current working directory.

This is one of the many features that will be exponentially nicer in a VSCode UX. 😐

Reply:
It may be better to save the cert file in the same location as `lwr.config.json` since we have another work item where we also need to update the config file to include cert & key. Bit I think in most (if not all) cases `process.cwd` would be the same as the location of `lwr.config.json`